SheetMetalFoldWall.py

```diff
@@ -0,0 +1,251 @@
+                    resultSolid = solid2
+                else :
+                    wallSolid = solid2
+                    resultSolid = soild1
+                    bendA = -bendA
+                    ExtrDir = -ExtrDir
+
+                # make sure the direction verctor is correct in respect to the normal
+                #print((normal.cross(revAxisV) - ExtrDir).Length)
+                if (normal.cross(revAxisV) - ExtrDir).Length < 1.0 :
+                    revAxisV = revAxisV * -1
+
+                if (len(cutFaceList)) > 1:
+                    cutFace = cutFaceList[0].multiFuse(cutFaceList[1:])
+
+                if not(flipped):
+                    revAxisP = sketchedge.valueAt(end) + normal * bendR
+                    revAxisV = revAxisV * -1
+                else :
+                    revAxisP = sketchedge.valueAt(end) + normal * -(bendR+thk)
+                print(revAxisP)
+
+                cutFace = cutFace.common(wallSolid)
+                cutLineList = []
+                for sketchEdge in sketch.Shape.Edges:
+                    if isinstance(sketchEdge.Curve, Part.Line):
+                        cutLine = sketchEdge.extrude(ExtrDir * actualLength)
+                        cutLineList.append(cutLine)
+                if (len(cutLineList)) > 1 :
+                    cutLine = cutLineList[0].multiFuse(cutLineList[1:])
+                else :
+                    cutLine = cutLineList[0]
+                BFace = cutLine.common(wallSolid)
+                for cutFaces in cutFaceList :
+                    cutExt = cutFaces.extrude(ExtrDir.normalize() * actualLength)
+                    wallSolid = wallSolid.cut(cutExt)
+                wallSolid.translate(ExtrDir * -actualLength)
+                wallSolid.rotate(revAxisP, revAxisV, bendA)
+                resultSolid = resultSolid.fuse(wallSolid)
+                bendSolid = cutFace.revolve(revAxisP, revAxisV, bendA)
+                #bendSolid = makeBend(BFace, bendR, bendA)
+                #Part.show(bendSolid)
+                resultSolid = resultSolid.fuse(bendSolid)
+        else :
+            if bendA > 0.0 :
+                # create bend
+                # FIXME undefined revFace!
+                bendSolid = revFace.extrude(revFace.normalAt(0,0) * unfoldLength)
+                #Part.show(bendSolid)
+                resultSolid = resultSolid.fuse(bendSolid)
+
+        Gui.ActiveDocument.getObject(MainObject.Name).Visibility = False
+        return resultSolid
+
+
+class SMFoldWall:
+    def __init__(self, obj):
+        """Add Wall with radius bend"""
+        selobj = Gui.Selection.getSelection()
+
+        obj.addProperty("App::PropertyLength","radius","Parameters","Bend Radius").radius = 1.0
+        obj.addProperty("App::PropertyBool","invert","Parameters","Invert bend direction").invert = False
+        obj.addProperty("App::PropertyAngle","angle","Parameters","Bend angle").angle = 90.0
+        # FIXME: I think this one does not work...
```

Comment:
there are three things with description "Invert bend direction": unfold, invert and invertbend.
Not sure which of them is the correct one or what the others do...